### PR TITLE
fix syncing contacts and closed groups after device linking

### PR DESCRIPTION
**Pull request by RyanRory**

### CODE REVIEW

File: src/org/thoughtcrime/securesms/jobs/MultiDeviceContactUpdateJob.java [hunk 1]
```diff
@@ -245,7 +245,7 @@ private void generateFullContactUpdate()
   }
 
   private List<ContactData> getAllContacts() {
-    List<Address> contactAddresses = DatabaseFactory.getRecipientDatabase(context).getRegistered();
+    List<Address> contactAddresses = new ArrayList<>(DatabaseFactory.getRecipientDatabase(context).getAllAddresses());
```
Comment (Mikunj) [inline src/org/thoughtcrime/securesms/jobs/MultiDeviceContactUpdateJob.java]: Does `getAllAddresses` filter out contacts we're not friends with and groups?

Comment (RyanRory) [reply src/org/thoughtcrime/securesms/jobs/MultiDeviceContactUpdateJob.java]: No, but we filter out the groups in the loop after that with `if (!address.isPhone()) { continue; }`. And we filter out those non-friend contacts in `generateFullContactUpdate()`

Comment (RyanRory) [reply src/org/thoughtcrime/securesms/jobs/MultiDeviceContactUpdateJob.java]: `getRegistered` just returns with an empty array, so I change to `getAllAddresses`...

Comment (Mikunj) [reply src/org/thoughtcrime/securesms/jobs/MultiDeviceContactUpdateJob.java]: 👍

**Pull request merged**